### DISCUSSION TITLE
Suggestion :  Offering in the same time both a READ as well as en EDIT mode in your table

### DISCUSSION THREAD

**Discussion by pycolle** [l-lin/angular-datatables#1566] [Ideas]
Hello 

I have one suggestion for you.

The idea is how to offer in the same time both a READ as well as en EDIT mode in your table .... 

Is it doable to have a sort of "FAKE EMPTY" first row in the table, which is always visible even if you move from one page to the other and which react the following : 
- Most of the time it will contain some empty EDIT or INPUTS field 
- if I click on one real row (may be on a button of this row (for asking edit), then EMPTY row become populated with the content of this current row and a sort of a UPDATE button appears allowing me to update the concerned row
- if the row is empty, then if I put something inside a SAVE button appears allowing me to SAVE (so to add one record in the database and subsequently in the table) 

I put an example attached, here the EMPTY row is at the bottom instead being at the top 
(and in this example I also have the famous problem which is resolved by hte DtTrigger on which I need to work)
In this example : my idea is that 
- the blue button on the right of each row will allow me to load the row content in the empty row  so that I can act on it .. 
- there is ten some Red, Yellow, Green buttons ... The red allow to delete, The Yellow to update and the Green to add
 (so green will be displayed only in case of new data enters, wheread red and yellow will be displayed after I clicked on 1 row)

![Suggestion_datatable_with_edit_row](https://user-images.githubusercontent.com/9412162/121057666-c99b2100-c7bf-11eb-9bb5-34352a292271.jpg)

**Comment by shanmukhateja** [comment]
Hi,

I don't know how would someone implement that. I would recommend looking into DataTables Editor library which provides in-line row editing and other goodies however is a commercial offering that . Also, your current approach could be difficult to work with from a UX perspective, 

**Note:** We do not offer support for DataTables Editor as we consider it out-of-scope for this library.

**Reply by pycolle** [reply]
Thanks a lot for the answer. 
I can understand for sure that it is not a small development at all as it need to be flexible depending on the table columns.
may be a suggestion. 
In my example I was able to add a row containing input areas as a simple row, the only catch is that it appears for the mopment at the end and working most surely because I xas not yet adding the famous dtTrigger ... 
May be you should do almost nothing except to see how such a manual line could systematically appear at first row ... there is may be a simple trick so that i behave this way ????

<table datatable class="row-border hover">
  <thead>
    <tr>
      <th witdh=0>n°</th>
      <th witdh=50> Category</th>
      <th witdh=50> Description</th>
      <th witdh=50> Direction </th>
      
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td><textarea rows="3" cols="20"></textarea></td>
      <td><input name="XX11" size=10></td>
      <td><input name="XX11" size=10></td>
      <td><input name="XX11" size=10></td>
      
    </tr>

    <tr *ngFor="let item of TheRequestLinksList ; let i = index">
      <td>{{i+1}}</td>
      <td>{{item._L_CATEG}}</td>
      <td>{{item._L_DESCR}}</td>
      <td>{{item._L_FLWDR}}</td>
      
      <td><button class="btn btn-primary" > * </button></td>
    </tr>
  </tbody>
</table>
